accelerators/federated-learning/automated_provisioning/ReadMe.md

```diff
@@ -0,0 +1,107 @@
+
+## Procedure
+
+> Set `accelerators/federated_learning/automated_provisioning` as your working directory!
+
+The procedure is fairly simple (detailed instructions are given in the below sections).
+1. After setting your working directory and  verifying you are meeting the prerequisites, you will first run the `CreateK8sCluster.ps1` script (with the appropriate arguments) and log in when prompted.
+2. Then you will run the `ConnectSiloToOrchestrator.ps1` script (with the appropriate arguments) and log in when prompted.
+
+We also suggest a way to run a simple validation job (highly recommended, to make sure the newly provisioned resources are usable). That being said, there is nothing special about this job, except for the fact that it will run on the new Arc-enabled K8s cluster; if you feel more comfortable using one of your own jobs to achieve that, it is perfectly acceptable.
+
+### Set up a silo
+
+For starters, you need to create a K8s cluster and the associated resource group if they don't exist already. Then you'll need to connect to this cluster and create the _kube_ config file (which will be used implicitly by the second script to point at this particular cluster). There is a script that does all that for you: `CreateK8sCluster.ps1`. It takes the following input arguments.
+- `SubscriptionId`: the Id of the subscription where the silo will be created (default: "48bbc269-ce89-4f6f-9a12-c6f91fcb772d", _a.k.a._ the AIMS subscription). 
+- `K8sClusterName`: the name of the K8s cluster to be created (default: "cont-k8s-01"). It will live in a resource group named like the cluster, with "-rg" appended.
+- `RGLocation`: the location of the K8s cluster and its corresponding resource group (default: "westus2").
+- `AgentCount`: the number of agents in the K8s cluster (default: 1 - beware, this should be an _int_, not a _string_).
+- `AgentVMSize`: The agent VM SKU (default: "Standard_B4ms").
+
+The command should look something like the below (with the parameters replaced by your own values of course).
+
+```ps1
+./ps/CreateK8sCluster.ps1 "Your-Silo-SubscriptionId" "Name-of-Cluster-to-Create" "Location-of-Cluster-to-Create" Number-of-Agents "VM-SKU"
+```
+
+### Connect it to the orchestrator workspace
+
+To connect a silo to an orchestrator Azure ML workspace, the following needs to happen:
+- create the Azure ML workspace if it doesn't already exist;
+- connect the K8s to Azure Arc; 
+- deploy the Azure ML extension on the Arc cluster;
+- attach the Arc cluster to the Azure ML workspace.
+
+Here again, there is a script that does all of that for you: `ConnectSiloToOrchestrator.ps1`. It takes the following input arguments.
+- `SubscriptionId_Orchestrator`: the Id of the subscription to which the orchestrator will belong (default: "48bbc269-ce89-4f6f-9a12-c6f91fcb772d", _a.k.a._ the AIMS subscription).
```

Comment:
Same "nit" about removing default value, ditto for e.g. workspace name, resource group, etc.